graylog2-server/src/main/java/org/graylog2/indexer/cluster/jest/JestUtils.java

```diff
@@ -84,6 +85,11 @@ private static ElasticsearchException specificException(Supplier<String> errorMe
                 case "index_not_found_exception":
                     final String indexName = asString(rootCause.get("resource.id"));
                     return buildIndexNotFoundException(errorMessage, indexName);
+                case "illegal_argument_exception":
+                    final String reason = asString(rootCause.get("reason"));
+                    if (reason != null && reason.startsWith("Expected numeric type on field")) {
+                        return buildFieldTypeException(errorMessage, reason);
+                    }
```

Comment:
Technically, we're missing a `break` here, which is omitted for the other cases because they return in every case, but not the `illegal_argument_exception` case.